soc/nxp/imx/imx8m/a53/soc.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/dt-bindings/rdc/imx_rdc.h>
+#include <fsl_common.h>
+#include <fsl_rdc.h>
+
+/* set RDC permission for peripherals */
+static void soc_rdc_init(void)
+{
+	rdc_domain_assignment_t assignment = {0};
+	rdc_periph_access_config_t periphConfig;
+
+	RDC_Init(RDC);
+	assignment.domainId = A53_DOMAIN_ID;
+	RDC_SetMasterDomainAssignment(RDC, kRDC_Master_A53, &assignment);
+
+	RDC_GetDefaultPeriphAccessConfig(&periphConfig);
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(uart2), okay)
```

Comment:
Don't configure RDC in case of no rdc property is available:

#if DT_NODE_HAS_STATUS(DT_NODELABEL(uart2), okay) && DT_NODE_HAS_PROP(DT_NODELABEL(uart2), rdc)

Reply:
make sense, will update. Thanks!